### ISSUE TITLE
Error when loading a saved stacking model

### ISSUE THREAD

**Issue by IncubatorShokuhou** [pycaret/pycaret#177]
I met an error when loading a saved stacking model  in tutorial REG102.

### To Reproduce

1. run REG102
2. restart the kernel
3. regenerate the dataframes 

```
from pycaret.datasets import get_data
from pycaret.regression import *
dataset = get_data('diamond')
data = dataset.sample(frac=0.9, random_state=786).reset_index(drop=True)
data_unseen = dataset.drop(data.index).reset_index(drop=True)
```

4. load saved experiment

`saved_experiment = load_experiment('Experiment_123 08Feb2020')`

5. reuse the model(according to the tutorial REG102

```
final_stack_1_loaded = saved_experiment[-1]
new_prediction = predict_model(final_stack_1_loaded, data=data_unseen, round = 0)
```
6. then the error is shown as below:

```
NameError                                 Traceback (most recent call last)
~/anaconda3/lib/python3.7/site-packages/pycaret/regression.py in predict_model(estimator, data, platform, authentication, round)
   6663 
-> 6664                 prep_pipe_transformer = prep_pipe
   6665                 target_transformer = target_inverse_transformer

NameError: name 'prep_pipe' is not defined

During handling of the above exception, another exception occurred:

SystemExit                                Traceback (most recent call last)
<ipython-input-6-48756811a733> in <module>
----> 1 new_prediction = predict_model(final_stack_1_loaded, data=data_unseen, round = 0)
      2 new_prediction.head()

~/anaconda3/lib/python3.7/site-packages/pycaret/regression.py in predict_model(estimator, data, platform, authentication, round)
   6669             except:
   6670 
-> 6671                 sys.exit("(Type Error): Transformation Pipeline Missing. ")
   6672 
   6673     else:

SystemExit: (Type Error): Transformation Pipeline Missing. 
```
Is there a bug or anything wrong with what I did?

**Comment by pycaret** [comment]
@IncubatorShokuhou Hi .. Your code doesn't look complete.  It seems like you haven't trained any model after loading data set.

**Comment by IncubatorShokuhou** [comment]
@pycaret maybe I didn't explain it clearly. I have trained the model in the tutorial REG102, and the model is saved into a pkl file at the end of this tutorial. Then I want to use the saved model in a new scrpit, but I failed to reuse it. I'm wondering what is wrong.

**Comment by pycaret** [comment]
@IncubatorShokuhou Its hard to answer without looking at notebook. Can you please share notebook so I can figure out what's wrong?

**Comment by IncubatorShokuhou** [comment]
@pycaret 
The notebook is your official [REG102 tutorial](https://github.com/pycaret/pycaret/blob/master/Tutorials/Regression%20Tutorial%20Level%20Intermediate%20-%20REG102.ipynb) :sweat_smile:

**Comment by saludes** [comment]
I have a similar problem: I start a new Python session with a saved experiment with two trained models, one of them at index 7:
```
from pycaret.regression import *
rpm = load_experiment('experiment_20200605_rpm')
predict_model(rpm[7])
```
Fails with 
```
NameError: name 'prep_pipe' is not defined
```
in about the same place.

But, the fact is that a toplevel variable named `prep_pipe` exists:
```
>>> prep_pipe
Pipeline(memory=None,
         steps=[('dtypes', ...
```

Looks like this variable is the one required in `load_experiment`but somehow, it isn't passed along.

**Comment by pycaret** [comment]
@saludes Apologies for the late reply. It somehow gets slipped from my radar. 

In PyCaret 2.0 we have got rid of `save_experiment` and `load_experiment`, instead, we are using `MLFlow` back-end now to provide end-to-end experiment logging. Please see this link for more details: https://www.pycaret.org/mlflow

If you are still looking for help, please feel free to open another issue.

**Comment by buddythumbs** [comment]
Hey - I had issues with this and found an answer here, the documentation has no mention of this anywhere - if you would like some additional contribution from community I would be happy to get involved.

**Comment by arenstar** [comment]
I still have issues.

I am producing a model as per shown in the examples however if i load the model in a new script and try to pass the same dataset to it ( shown below ) i get the error: **NameError: name 'prep_pipe' is not defined**

```
import pandas as pd
dataset = pd.read_csv('./out.csv')
#get last row from dataset
singleton = dataset.iloc[[-1]]
columns = singleton.columns
data = singleton.iloc[-1].values.tolist()

data_unseen = pd.DataFrame([data], columns = columns)
data_unseen.drop('result', axis = 1, inplace = True)

from pycaret.regression import *
model = load_model('Final_Best')
predict_model(model, data=data_unseen, round = 0)
```
Error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/Caskroom/miniforge/base/envs/ml/lib/python3.8/site-packages/pycaret/regression.py", line 1850, in predict_model
    return pycaret.internal.tabular.predict_model(
  File "/opt/homebrew/Caskroom/miniforge/base/envs/ml/lib/python3.8/site-packages/pycaret/internal/tabular.py", line 8938, in predict_model
    dtypes = estimator.named_steps.get("dtypes", prep_pipe.named_steps["dtypes"])
NameError: name 'prep_pipe' is not defined
```


However if i add `exp_clf101 = setup(data = dataset, target = 'result', use_gpu=False, silent=True)` after loading the model it works.

**Comment by joanlofe** [comment]
> However if i add `exp_clf101 = setup(data = dataset, target = 'result', use_gpu=False, silent=True)` after loading the model it works.

@arenstar I can confirm this works. And it is true that this is not mentioned anywhere in the documentation, or at least I've been not able to see it.

**Comment by vverlaine** [comment]
could not be resolved?

**Comment by ngupta23** [comment]
Are you trying this with the latest version of pycaret? 

Can you paste a reproducible example of the issue you are facing (data + code)

📌 Overview: http://ow.ly/LoIv50IL5RQ
📌 Examples (Do's and Dont's): http://ow.ly/AXKm50IL5RR

Also, I would recommend starting a new issue instead of reviving this old one. If you think this is related, you can reference this in the new issue.

**Comment by estebanarr** [comment]
Same issue here after using predict_model: NameError: name 'prep_pipe' is not defined"

**Comment by ngupta23** [comment]
This should be fixed in the latest version 2.3.10
